operations/document-compression-updater/updater.py

```diff
@@ -0,0 +1,271 @@
+import datetime
+import sys
+import random
+import json
+import pymongo
+import time
+import threading
+import os
+import multiprocessing as mp
+import argparse
+import string
+import math
+
+
+def deleteLog(appConfig):
+    if os.path.exists(appConfig['logFileName']):
+        os.remove(appConfig['logFileName'])
+
+def printLog(thisMessage,appConfig):
+    print("{}".format(thisMessage))
+    with open(appConfig['logFileName'], 'a') as fp:
+        fp.write("{}\n".format(thisMessage))
+
+
+
+def setup(appConfig):
+
+    databaseName = appConfig['databaseName']
+    collectionName = appConfig['collectionName']
+
+    client = pymongo.MongoClient(appConfig['uri'])
+    
+    # database and collection for compression
+
+    db = client[databaseName]
+    adminDb = client['admin']
+    col = db[collectionName]
+    
+    # database and collection for tracking
+
+    tracker_db=client['tracker_db']
+    trackerCollectionName = collectionName+'_tracker_col'
+    tracker_col=tracker_db[trackerCollectionName]
+    
+    list_of_collections = tracker_db.list_collection_names()  # Return a list of collections in 'tracker_db'
+    print("list_of_collections {}".format(list_of_collections))
+    
+    if trackerCollectionName in list_of_collections :
+    
+        # tracker db already has entry for collection
+
+        result = tracker_col.find({}).sort({ "_id" : -1}).limit(1)   
+               
+        for lastEntry in result :
+            numExistingDocuments = lastEntry["numExistingDocuments"]
+            maxObjectIdToTouch = lastEntry["maxObjectIdToTouch"]
+            lastScannedObjectId = lastEntry["lastScannedObjectId"]
+            numDocumentsUpdated = lastEntry["numDocumentsUpdated"]
+            print("Found existing record: {}".format(str(lastEntry))) 
+
+    else :      
+    
+        # create first entry in tracker db  for collection
+        result = col.find({},{ "_id" :1}).sort({ "_id" :-1}).limit(1)
+        for id in result :
+            print("result {}".format(result))
+            maxObjectIdToTouch = id["_id"]
+        lastScannedObjectId = 0
+        numDocumentsUpdated = 0
+           
+        numExistingDocuments = col.estimated_document_count()
+        
+            
+        first_entry = {
+            "collection_name": appConfig['collectionName'],
```

Comment:
How are you handling the use-case where the same collection name exists in multiple databases?

Reply:
agree, i also need to add the db name